src/main/bootstrap/mod.rs

```diff
@@ -455,44 +456,36 @@ mod tests {
 
             let (tx, rx) = mpsc::channel();
 
-            let duplicates: Vec<PeerInfo> = peers.iter().take(5).map(|peer| peer.clone()).collect();
+            let duplicates: HashSet<PeerInfo> =
+                peers.iter().take(5).map(|peer| peer.clone()).collect();
             tx.send(duplicates).unwrap();
 
-            update_peers(&mut peers, rx);
+            receive_peers_into(&mut peers, rx);
 
-            let peers_len = peers.len();
-            let mut peers_set = HashSet::new();
+            let peers_set: HashSet<PeerInfo> = peers.iter().cloned().collect();
 
-            assert!(peers.into_iter().all(|peer| peers_set.insert(peer)));
-            assert_eq!(peers_len, peers_set.len());
+            assert_eq!(peers.len(), peers_set.len());
         }
 
         #[test]
-        fn it_preserves_order_if_shuffled_again() {
-            let hard_coded_peer = peer_info_with_rand_key(ipv4_addr(1, 2, 3, 4, 4000));
-            let cached_peers: HashSet<PeerInfo> = (0..10)
+        fn it_preserves_order_of_old_peers() {
```

Comment:
Could you add one more test to see if the peers from `rx` are actually added? Now we have tests to see if 
1. duplicates are omitted 
2. order is preserved
but there's no test that checks if peers from `rx` are added :/

Reply:
Sure, done.